<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Project Code Line Counter</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="2,6,0,2,0,0,8,1">
   <property name="spacing">
    <number>6</number>
   </property>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_4" stretch="2,1">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0,0">
       <property name="spacing">
        <number>7</number>
       </property>
       <item>
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Project Dir:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>50</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item alignment="Qt::AlignHCenter">
      <widget class="QLabel" name="label_6">
       <property name="font">
        <font>
         <family>Arial</family>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>All Right</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxOptions">
     <property name="title">
      <string>Counting Options</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>1</number>
      </property>
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>General</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QScrollArea" name="scrollArea">
            <property name="widgetResizable">
             <bool>true</bool>
            </property>
            <widget class="QWidget" name="scrollAreaWidgetContents">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>0</y>
               <width>755</width>
               <height>366</height>
              </rect>
             </property>
             <layout class="QGridLayout" name="gridLayout_3">
              <item row="0" column="1">
               <widget class="QGroupBox" name="groupBox">
                <property name="title">
                 <string>Dir BlackList</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_3">
                 <item>
                  <widget class="QListWidget" name="blacklist">
                   <property name="maximumSize">
                    <size>
                     <width>16777215</width>
                     <height>16777215</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,1,6">
                   <property name="spacing">
                    <number>17</number>
                   </property>
                   <item>
                    <widget class="QToolButton" name="toolButton_2">
                     <property name="text">
                      <string/>
                     </property>
                     <property name="icon">
                      <iconset resource="resources/main.qrc">
                       <normaloff>:/add-select.svg</normaloff>:/add-select.svg</iconset>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QToolButton" name="toolButton">
                     <property name="text">
                      <string/>
                     </property>
                     <property name="icon">
                      <iconset resource="resources/main.qrc">
                       <normaloff>:/sami-select.svg</normaloff>:/sami-select.svg</iconset>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <spacer name="horizontalSpacer_3">
                     <property name="orientation">
                      <enum>Qt::Horizontal</enum>
                     </property>
                     <property name="sizeHint" stdset="0">
                      <size>
                       <width>40</width>
                       <height>20</height>
                      </size>
                     </property>
                    </spacer>
                   </item>
                  </layout>
                 </item>
                </layout>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QCheckBox" name="checkBox">
                <property name="text">
                 <string>Count Sub Dir of Project</string>
                </property>
                <property name="checkable">
                 <bool>true</bool>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QGroupBox" name="groupBox_4">
                <property name="title">
                 <string>Extensions BlackList</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <widget class="QLineEdit" name="lineEdit_ext"/>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_22">
                   <property name="text">
                    <string>All Right</string>
                   </property>
                  </widget>
                 </item>
                 <item alignment="Qt::AlignTop">
                  <widget class="QLabel" name="label_21">
                   <property name="text">
                    <string>Enter several extensions separated by commas.For example:exe,png,dll</string>
                   </property>
                   <property name="wordWrap">
                    <bool>true</bool>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QCheckBox" name="checkBox_2">
                <property name="text">
                 <string>BlackList for FileExt:</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_2">
         <attribute name="title">
          <string>Total Chars</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QCheckBox" name="checkBox_3">
            <property name="text">
             <string>Blank</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_3">
         <attribute name="title">
          <string>Total Lines</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QCheckBox" name="checkBox_6">
            <property name="text">
             <string>Blank Line</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <property name="spacing">
             <number>0</number>
            </property>
            <item>
             <widget class="QCheckBox" name="checkBox_7">
              <property name="text">
               <string/>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_3">
              <property name="text">
               <string>Do not count Lines with less than</string>
              </property>
              <property name="flat">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinBox">
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="pushButton_4">
              <property name="text">
               <string>character(s)</string>
              </property>
              <property name="flat">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>472</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="pushButton_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>20</pointsize>
       <weight>50</weight>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>Count</string>
     </property>
     <property name="default">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item alignment="Qt::AlignHCenter">
    <widget class="QLabel" name="label_loading">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;Counting...&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Result</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="2,1">
      <item>
       <widget class="QTableView" name="tableView">
        <property name="gridStyle">
         <enum>Qt::SolidLine</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_5" rowstretch="1,1">
        <item row="0" column="1">
         <widget class="QLabel" name="label_resultA">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(227, 227, 227);
padding-left: 5px;</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="label_resultB">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(227, 227, 227);
padding-left: 5px;</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Total Files:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Total Chars:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label_resultC">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(227, 227, 227);
padding-left: 5px;</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Total Lines:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="label_resultD">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(227, 227, 227);
padding-left: 5px;</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Total Lines Without</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_6">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">text-decoration:underline;
margin: 0</string>
            </property>
            <property name="text">
             <string>Comments:</string>
            </property>
            <property name="flat">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string/>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="9,0">
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Minimum</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_5">
        <property name="text">
         <string>Clear Result</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources/main.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>lineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>Widget</receiver>
   <slot>timeOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>168</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>788</x>
     <y>33</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>blacklistAdd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>blacklistDel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>268</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>312</x>
     <y>125</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_2</sender>
   <signal>clicked(bool)</signal>
   <receiver>groupBox_4</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>180</x>
     <y>433</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>407</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_ext</sender>
   <signal>editingFinished()</signal>
   <receiver>Widget</receiver>
   <slot>timeOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>391</x>
     <y>423</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEdit_ext</sender>
   <signal>cursorPositionChanged(int,int)</signal>
   <receiver>Widget</receiver>
   <slot>extBlackListChanged(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>426</x>
     <y>423</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>197</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>startCount()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>449</x>
     <y>343</y>
    </hint>
    <hint type="destinationlabel">
     <x>449</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_7</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>34</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_7</sender>
   <signal>toggled(bool)</signal>
   <receiver>pushButton_3</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>34</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBox_7</sender>
   <signal>toggled(bool)</signal>
   <receiver>pushButton_4</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>34</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>356</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>checkBox_7</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>34</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>checkBox_7</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>356</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>34</x>
     <y>203</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>clearResult()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>740</x>
     <y>660</y>
    </hint>
    <hint type="destinationlabel">
     <x>972</x>
     <y>592</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>showComments()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>541</x>
     <y>577</y>
    </hint>
    <hint type="destinationlabel">
     <x>1200</x>
     <y>613</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>timeOut()</slot>
  <slot>openFile()</slot>
  <slot>blacklistAdd()</slot>
  <slot>blacklistDel()</slot>
  <slot>extBlackListChanged(int,int)</slot>
  <slot>startCount()</slot>
  <slot>clearResult()</slot>
  <slot>showComments()</slot>
 </slots>
</ui>
